sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -181,13 +181,20 @@ export class AppConfigurationClient {
       "AppConfigurationClient.addConfigurationSetting",
       options,
       async (updatedOptions) => {
+        let status;
         const keyValue = serializeAsConfigurationSettingParam(configurationSetting);
         logger.info("[addConfigurationSetting] Creating a key value pair");
         const originalResponse = await this.client.putKeyValue(configurationSetting.key, {
           ifNoneMatch: "*",
           label: configurationSetting.label,
           entity: keyValue,
-          ...updatedOptions,
+          ...updatedOptions,          
+          onResponse: (response) => {
+            status = response.status;
+            if (status === 412){
+              throw new Error ("Setting was already present");
```

Comment:
1. Throw a RestError instead
2. Perhaps, update the actual error object instead of creating a new one?